src/pkg/packager/migrations/common.go

```diff
@@ -1,80 +1,71 @@
 	mitigation string
 }
 
-// List of migrations tracked in the zarf.yaml build data.
-const (
-	// This should be updated when a breaking change is introduced to the Zarf package structure.  See: https://github.com/defenseunicorns/zarf/releases/tag/v0.27.0
-	LastNonBreakingVersion   = "v0.27.0"
-	ScriptsToActionsMigrated = "scripts-to-actions"
-	PluralizeSetVariable     = "pluralize-set-variable"
-)
+// LastNonBreakingVersion is the last version that did not have any breaking changes
+//
+// This should be updated when a breaking change is introduced to the Zarf package structure.  See: https://github.com/defenseunicorns/zarf/releases/tag/v0.32.2
+const LastNonBreakingVersion = "v0.27.0"
 
 // List of breaking changes to warn the user of.
-var breakingChanges = []BreakingChange{
+var breakingChanges = []breakingChange{
 	{
 		version:    semver.New(0, 26, 0, "", ""),
 		title:      "Zarf container images are now mutated based on tag instead of repository name.",
 		mitigation: "Reinitialize the cluster using v0.26.0 or later and redeploy existing packages to update the image references (you can view existing packages with 'zarf package list' and view cluster images with 'zarf tools registry catalog').",
 	},
 }
 
-// MigrateComponent runs all migrations on a component.
-// Build should be empty on package create, but include just in case someone copied a zarf.yaml from a zarf package.
-func MigrateComponent(build types.ZarfBuildData, component types.ZarfComponent) (migratedComponent types.ZarfComponent, warnings []string) {
-	migratedComponent = component
-
-	// If the component has already been migrated, clear the deprecated scripts.
-	if slices.Contains(build.Migrations, ScriptsToActionsMigrated) {
-		migratedComponent.DeprecatedScripts = types.DeprecatedZarfComponentScripts{}
-	} else {
-		// Otherwise, run the migration.
-		var warning string
-		if migratedComponent, warning = migrateScriptsToActions(migratedComponent); warning != "" {
-			warnings = append(warnings, warning)
-		}
-	}
+// PackageOrComponent is a type that can be either a ZarfComponent or a ZarfPackage
+type PackageOrComponent interface {
+	types.ZarfComponent | types.ZarfPackage
+}
 
-	// If the component has already been migrated, clear the setVariable definitions.
-	if slices.Contains(build.Migrations, PluralizeSetVariable) {
-		migratedComponent = clearSetVariables(migratedComponent)
-	} else {
-		// Otherwise, run the migration.
-		var warning string
-		if migratedComponent, warning = migrateSetVariableToSetVariables(migratedComponent); warning != "" {
-			warnings = append(warnings, warning)
-		}
-	}
+// Migration represents a migration on types satisfying the PackageOrComponent interface.
+type Migration[T PackageOrComponent] interface {
+	fmt.Stringer
+	Run(T) (T, string)
+}
+
+// DeprecatedMigration represents a migration on types satisfying the PackageOrComponent interface.
+type DeprecatedMigration[T PackageOrComponent] interface {
+	Migration[T]
+	Clear(T) T
```

Comment:
This interface feels a bit weird to me, I don't like that it's easy to clear without running first. For example in the readZarfYaml function we only clear if the migrations are set, whereas on main currently we will migrate. We also don't clear scripts after the actions are migrated, though that behavior is the same as main, but I think it should be changed and tested.  

I can't think of any situations where we would want to clear without running first, I think it'd probably be fine if we only had one migration interface and the deprecated migrations also cleared on run, but I'm open to other ideas.